src/components/UASTViewer.js

```diff
@@ -1,18 +1,35 @@
 import React, { Component } from 'react';
 import styled from 'styled-components';
+import Node from './uast/Node';
+import {eventHandler} from './uast/EventHandler';
+import {font, background} from '../styling/variables';
 
 const Container = styled.div`
+  position: absolute;
+  height: 100%;
+  width: 100%;
+  overflow: auto;
+  font-family: ${font.family.code};
+  font-size: ${font.size.large};
+  background: ${background.main};
+  line-height: ${font.lineHeight.large};
+  padding: 4px;
 `
 
 export default class UASTViewer extends Component {
+  constructor(props) {
+    super(props);
+    eventHandler.onNodeSelected = this.props.onNodeSelected;
```

Comment:
It is not a singleton but a class instantiated only once.
I prefer much more anything closer to throwing events, than passing callback deeper and deeper everywhere, but I can refactor this piece for sure